packages/common/src/ui/layout/tables/columns/NameAndColorColumn.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { CellProps, Column } from 'react-table';
+import { TableCell } from '@material-ui/core';
+import { Color, ColorMenu, UnstyledIconButton } from '../../../components';
+import { Circle } from '../../../icons';
+import { Box } from '@material-ui/system';
+
+type Colorable = {
+  color: string;
+} & Record<string, unknown>;
+
+export const getNameAndColorColumn = <T extends Colorable>(
+  onChange: (row: T, color: Color) => void
+): Column<T> => ({
```

Comment:
Was going to create an interface for `ColumnFactory` .. but wanted to see what you thought about this approach rather than the other options